### ISSUE TITLE
'\t' character in string throws a DocumentDrawingException with Arial font.

### ISSUE THREAD

**Issue by L-E-Freeman** [QuestPDF/QuestPDF#584]
**Describe the bug**
Attempting to display a string containing a `\t` in a table cell causes a DocumentDrawingException. [This](https://www.fileformat.info/info/unicode/font/arial/list.htm) source says that the Arial font should support the `\t` character. The exception thrown is the following: 

```
QuestPDF.Drawing.Exceptions.DocumentDrawingException: Could not find an appropriate font fallback for glyph: U-0009 '\t'. Font families available on current environment that contain this glyph: Microsoft PhagsPa. Possible solutions: 1) Use one of the listed fonts as the primary font in your document. 2) Configure the fallback TextStyle using the 'TextStyle.Fallback' method with one of the listed fonts.
```

Setting the fallback font to `Microsoft PhagsPa` does stop the exception, but that's a font for typing Mongolian characters and isn't really a solution since Arial should support this character.

**To Reproduce**

I set the default page font to Arial in the `Compose()` method like this: 

```
public void Compose (IDocumentContainer container)
        {
            container
                .Page(page =>
                {
                    page.DefaultTextStyle(x => x
                        .FontFamily(Fonts.Arial));

                    // Drawing the page content, table with problem is drawn in ComposeContent()
                    page.Header().ShowOnce().Element(ComposeHeader);
                    page.Content().Element(ComposeContent);
                    page.Footer().Element(ComposeFooter);
                });
        }
```

Then I have a table cell that looks like this: 

```
table.Cell().Element(ContentCellStyle).AlignCenter().Text(item.theProblemString);
```

```
static IContainer ContentCellStyle(IContainer container) {
    return container.DefaultTextStyle(text => text.FontSize(8)).PaddingVertical(1);
}
```

`item.theProblemString` is a string property of my class instance that contains a `\t` character.

**Expected behavior**
An exception should not be thrown when a `\t` character appears in a string while the Arial font is used.

**Screenshots**
N/A

**Additional context**
N/A

**Comment by y0ung3r** [comment]
After a little research, I don't think it's a bug. `Skia` just doesn't know how to render `\t` character (like all "empty" characters). This applies to any font, by the way

**Comment by y0ung3r** [comment]
If the problem exists, it is definitely on the `Skia` side. Somewhere in here https://github.com/google/skia/blob/ebae76679a04e9c8666680b2f439fa88624971fd/src/core/SkTypeface.cpp#L100C13-L100C13

**Comment by MarcinZiabek** [comment]
Thank you so much for investigating this issue! I highly appreciate your help. 

I believe that the problem could be more subtle than we think. Indeed, the font may define the tab `\t` codepoint but at the same time may not contain its visual representation - and therefore cause the error. At the same time, SkiaSharp has limited text rendering capabilities. Maybe more sophisticated features are on the client side, for example: shaping, Unicode standard and special characters (such as new-line `\n` and tab `\t`) support.

In such a case, QuestPDF should somehow handle the `\t` character to prevent runtime errors. There are a couple of potential strategies:
1) ignore this character (I believe this is done by most of the libraries),
2) replace it with an arbitrary number of spaces,
3) replace it with an empty space of specific size,
4) implement proper support.

I believe that options 2) and 3) are relatively close to the desired behavior, especially when the '\t' character is at the very beginning of the line. While option 1) is the most straightforward.

In the future, this problem may be solved entirely as part of the https://github.com/QuestPDF/QuestPDF/discussions/622 effort. So even temporary improvements could be beneficial.

What do you think?

**Comment by y0ung3r** [comment]
@MarcinZiabek Thank you for your detailed response! Absolutely agree with everything you wrote

> ignore this character (I believe this is done by most of the libraries)

If we're not just talking about `PDF`, I've seen libraries (like `DocX`) that support the `\t` glyph. But the problem is that `OpenXML`, unlike `PDF`, knows how to handle `\t` by default with a special element called `tab`

> In such a case, QuestPDF should somehow handle the \t character to prevent runtime errors

We can look at `OpenXML` and do something like this: http://officeopenxml.com/WPtab.php. I mean `\t` glyph should be treated as a separate element within `TextBlock`

**Comment by MarcinZiabek** [comment]
Indeed, the support in OpenXML is trivial as the rendering process is entirely handled by the Word processor.

The difficulty here is that the tab glyph is not a visual character but a particular corner case that needs to be handled explicitly in the text rendering code. It can be as simple as inserting an empty space, whose size is dependent on the configured tab size and the total horizontal size of all text runs before the tab in the current line. https://en.wikipedia.org/wiki/Tab_key

But my experience says that nothing related to text rendering is simple 🤣

So the only question is, how many users require this feature and at what level?

**Comment by MarcinZiabek** [comment]
In the 2024.3 release, we introduced a redesigned text rendering engine with significantly enhanced capabilities. It now supports the vast majority of use cases, including better support for the `/t` character.

Since there has been no recent activity, I’m closing this issue. If you have a similar request in the future, please feel free to open a new GitHub issue. Thanks for your feedback! ❤️
